code/game/objects/items/weapons/improvised_components.dm

```diff
@@ -69,3 +69,43 @@
 		qdel(src)
 		user.put_in_hands(finished)
 	update_icon(user)
+
+/obj/item/weapon/material/shaft
+	name = "shaft"
+	desc = "A large stick, you could probably attach something to it."
+	icon_state = "shaft"
+	item_state = "rods"
+	force = 5
+	throwforce = 3
+	w_class = 4
+	attack_verb = list("hit", "bludgeoned", "whacked", "bonked")
+	force_divisor = 0.1
+	thrown_force_divisor = 0.1
+	default_material = "wood"
+
+/obj/item/weapon/material/shaft/attackby(var/obj/item/I, mob/user as mob)
+	..()
+	var/obj/item/finished
+	if(istype(I, /obj/item/weapon/material/spearhead))
+		var/obj/item/weapon/material/spearhead/tip = I
+		finished = new /obj/item/weapon/material/twohanded/pike(get_turf(user), tip.material.name)
+		user << "<span class='notice'>You attach \the [I] to the top of \the [src].</span>"
+	if(finished)
+		user.drop_from_inventory(src)
+		user.drop_from_inventory(I)
+		qdel(I)
+		qdel(src)
+		user.put_in_hands(finished)
+	update_icon(user)
+
+/obj/item/weapon/material/spearhead
+	name = "spearhead"

```

Comment:
This seems a little deceptive, considering it makes a pike and not a spear - both objects that exist.

Reply:
Any another suggestion for another name?

Reply:
Fixed it by allowing people to also make spears using spearheads.